src/hooks/shared.ts

```diff
@@ -95,3 +95,37 @@ export const onComplete = <E extends HTMLElement | SVGElement>({
 
   onAnimationComplete?.();
 };
+
+interface CheckCacheParams {
+  cache: MutableRefObject<AnimationCache>;
+  index: number;
+  from: CSSProperties;
+  to: CSSProperties;
+}
+
+/**
+ * The checkAnimationCache functions inspects the hook's local animation cache to see
+ * if there are animation properties applied. It also checks if the cached properties
+ * match that of the to configuration, signaling it's safe to animate. Otherwise, use
```

Comment:
nit: typo
`match that of the to configuration`

Reply:
This is intentional language, but I see how its confusing. I'll put `to` in quotes to match it clear (it's referencing the `to` property specified in a `renature` hook).

Reply:
OH right I see what you mean